tests/e2e/module_install_by_version_test.go

```diff
@@ -0,0 +1,50 @@
+package e2e_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+	"github.com/kyma-project/lifecycle-manager/pkg/testutils/builder"
+)
+
+var _ = Describe("Module Install By Version", Ordered, func() {
+	kyma := NewKymaWithSyncLabel("kyma-sample", ControlPlaneNamespace, v1beta2.DefaultChannel)
+	templateOperatorModule := NewTemplateOperatorWithVersion("2.4.2-e2e-test")
+	moduleCR := builder.NewModuleCRBuilder().
+		WithName("sample-2.4.2-e2e-test").
+		WithNamespace(RemoteNamespace).Build()
+
+	InitEmptyKymaBeforeAll(kyma)
+	CleanupKymaAfterAll(kyma)
+
+	Context("Given SKR Cluster", func() {
+		It("When Template-Operator Module is enabled on SKR Kyma CR in a specific version", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(runtimeClient, defaultRemoteKymaName, RemoteNamespace, templateOperatorModule).
+				Should(Succeed())
+		})
+
+		It("Then Module CR exists", func() {
+			Eventually(ModuleCRExists).
+				WithContext(ctx).
+				WithArguments(runtimeClient, moduleCR).
+				Should(Succeed())
+
+			By("And Module Operator Deployment exists")
+			Eventually(DeploymentIsReady).
+				WithContext(ctx).
+				WithArguments(runtimeClient, ModuleDeploymentNameInNewerVersion, TestModuleResourceNamespace).
+				Should(Succeed())
+
+			By("And KCP Kyma CR is in \"Ready\" State")
+			Eventually(KymaIsInState).
```

Comment:
it's better also verify module status is expected

Reply:
I'll add this Expectation

Reply:
Added